eng/common/TestResources/New-TestResources.ps1

```diff
@@ -313,25 +312,63 @@ foreach ($templateFile in $templateFiles) {
         }
     }
 
-    foreach ($key in $deploymentOutputs.Keys)
+    if ($OutFile)
     {
-        $value = $deploymentOutputs[$key]
-        
-        if ($CI) {
-            # Treat all ARM template output variables as secrets since "SecureString" variables do not set values.
-            # In order to mask secrets but set environment variables for any given ARM template, we set variables twice as shown below.
-            Write-Host "Setting variable '$key': ***"
-            Write-Host "##vso[task.setvariable variable=_$key;issecret=true;]$($value)"
-            Write-Host "##vso[task.setvariable variable=$key;]$($value)"
-        } else {
-            Write-Host ($shellExportFormat -f $key, $value)
+        if (!$IsWindows)
+        {
+            Write-Host "File option is supported only on Windows"
         }
+
+        # Compute a hash of repository root to scope the test environment file
+        $repositoryRootHash = New-Object System.Text.StringBuilder
+        [System.Security.Cryptography.HashAlgorithm]::Create("SHA1").ComputeHash([System.Text.Encoding]::UTF8.GetBytes($repositoryRoot)) | `
+            %{ [Void]$repositoryRootHash.Append($_.ToString("x2")) }
+
+        $outputPath = Join-Path $env:USERPROFILE ".Azure" "TestEnvironments" $repositoryRootHash
```

Comment:
Should we put this next to the arm template? To make clean-up a little easier? 

We should add some clean-up logic in the remove resource script to delete this file.

Reply:
we support multiple ARM templates under the same service so it's not clear which one to have it next to.

Reply:
Looked again, making the change.